bootstrap.sh

```diff
@@ -29,6 +29,14 @@ install_manifest() {  # $1 = name of the manifest file
     curl --location --remote-name https://raw.githubusercontent.com/$MANIFESTS_SOURCE_REPOSITORY/$MANIFESTS_SOURCE_FOLDER/$1.pp --output $BOOTSTRAP_MANIFESTS_DESTINATION_FOLDER
 }
 
+function run_puppet {
+    set +e
+    puppet apply --detailed-exitcodes --verbose $@
+    exit_code=$?
+    set -e
+    [ $exit_code -eq 0  ] || [ $exit_code -eq 2 ]
```

Comment:
> This could also be written `[[ $exit_code == 0 ]]`: no need to test for numeric equality with `-eq`, string equality is enough and more readable, and `[[ ]]` means no subprocess and thus better performance.